packages/calcite-components/src/components/tree/tree.tsx

```diff
@@ -228,7 +228,7 @@ export class Tree {
     } else if (!isNoneSelectionMode) {
       targetItems.forEach((treeItem) => {
         if (!treeItem.disabled) {
-          treeItem.selected = true;
+          treeItem.selected = this.selectionMode === "single" ? !treeItem.selected : true;
```

Comment:
Suggestion: `this.selectionMode === "single" || !treeItem.selected;`

Reply:
Actually, the two lines are not logically equivalent: The suggested line will return "true" whenever "selectionMode" is "single"; however, the existing line can return true or false when "selectionMode" is "single". The line can be changed to:

`this.selectionMode !== "single" || !treeItem.selected"`

Reply:
Changes look good. 👍 Merge approved